rules.bzl

```diff
@@ -13,7 +15,7 @@ def _linter_impl(ctx):
     return [
         LinterInfo(
             executable_path=ctx.attr.executable_path,
-            executable=ctx.executable.executable,
+            executable=ctx.attr.executable,
```

Comment:
I suppose either way it's the same `File` object? What's the reason for the change?

Reply:
No, we're changing it from a [`File`](https://docs.bazel.build/versions/master/skylark/lib/File.html) object (`ctx.executable.executable`) to a [`Target`](https://docs.bazel.build/versions/master/skylark/lib/Target.html) object. You will find that `ctx.attr.executable[DefaultInfo].files_to_run.executable == ctx.executable.executable`.

We're using this with `clang-format` and `cpplint`. With `clang-format` you have a binary that needs at least two shared objects to be available in the sandbox. We have `cpplint` in a `py_binary`, and then it wanted an entire runfiles structure. In either case, we need more than just a file. We also need the runtime dependencies of the executable. We solve this with `ctx.resolve_tool(tools = [linter[LinterInfo].executable])` (see [ctx.resolve_tools](https://docs.bazel.build/versions/master/skylark/lib/ctx.html#resolve_tools)), which requires a sequence of  [`Target`](https://docs.bazel.build/versions/master/skylark/lib/Target.html) objects.